File: ruby/reverse_string.rb
def reverse_string(str)
  # newArr = str.split("")
  newStr = ""
  str.chars.each{|letter| newStr = letter + newStr}
  newStr
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts 

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/reverse_string.js
function reverseString(str) {
  newArr = str.split("")
  let reverseArr = []
  for(i=newArr.length-1; i >= 0; i--) {
    reverseArr.push(newArr[i])
  }
  return reverseArr.join("")
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution
